core/unix/injector.c

```diff
@@ -1490,7 +1492,12 @@ inject_ptrace(dr_inject_info_t *info, const char *library_path)
         if (r < 0 || !WIFSTOPPED(status))
             return false;
         signal = WSTOPSIG(status);
-    } while (signal == SIGSEGV);
+    } while (signal == SIGSEGV || signal == SIGILL);
+    /* SIGILL has a handler at signal_arch_init and defined to XSTATE_QUERY_SIG (x86)
+     * or VFP_QUERY_SIG (arm). It will dispatch by thread_signal in signal_arch_init.
+     * But if the child process is ptraced, the signal will catch by parent. We just
+     * pass it back to child.
```

Comment:
I meant to just update the existing comment on lines 1482-1483 which talks about passing SIGSEGV: just add a short sentence about also passing SIGILL due to signal_arch_init. The comment you added here *after* the SIGILL code and separate from the existing SIGSEGV comment may confuse the reader due to its placement.